File: src/java/Power.java
import java.math.BigInteger;

public class Power {
    public static void main(String[] args) {
        String baseEnv = System.getenv("BASE");
        String expEnv = System.getenv("EXP");
        
        int base = 2;
        int exp = 1000;
        
        if (baseEnv != null) {
            try {
                base = Integer.parseInt(baseEnv);
            } catch (NumberFormatException e) {
                // Keep default
            }
        }
        
        if (expEnv != null) {
            try {
                exp = Integer.parseInt(expEnv);
            } catch (NumberFormatException e) {
                // Keep default
            }
        }

        BigInteger bigBase = BigInteger.valueOf(base);

        long start = System.nanoTime();
        BigInteger result = bigBase.pow(exp);
        long end = System.nanoTime();
        double durationMs = (end - start) / 1_000_000.0;

        System.out.println("Result(" + base + "^" + exp + "): " + result);
        System.out.printf("Time: %.3f ms%n", durationMs);
    }
}

File: src/js/power.js
const { performance } = require('perf_hooks');

const base = BigInt(parseInt(process.env.BASE) || 2);
const exp = parseInt(process.env.EXP) || 1000;

// Binary exponentiation for BigInt
function power(base, exp) {
    let result = 1n;
    let b = base;
    let e = exp;

    while (e > 0) {
        if (e % 2 === 1) {
            result *= b;
        }
        b *= b;
        e = Math.floor(e / 2);
    }

    return result;
}

const start = performance.now();
const result = power(base, exp);
const end = performance.now();

console.log(`Result(${base}^${exp}): ${result.toString()}`);
console.log(`Time: ${(end - start).toFixed(3)} ms`);
